src/source/source_cache.js

```diff
@@ -164,6 +170,22 @@ class SourceCache extends Evented {
         }
 
         this._state.coalesceChanges(this._tiles, this.map ? this.map.painter : null);
+
+        if (this._source.prepareTile) {
+            for (const i in this._tiles) {
+                const tile = this._tiles[i];
+                const data = this._source.prepareTile(tile);
```

Comment:
`data` doesn't appear to be used elsewhere so can we avoid allocating the variable?

Reply:
I'm using `data` to check if we need to clear the terrain render cache

https://github.com/mapbox/mapbox-gl-js/blob/fcbf0b97f6eaf7de3f2fc37f86ce9d45e0caa82a/src/source/source_cache.js#L177-L180